File: 1552-build-an-array-with-stack-operations/1552-build-an-array-with-stack-operations.java
class Solution {
    public List<String> buildArray(int[] target, int n) {
        List<String> ret = new ArrayList<>();
        int val = 1;
        for (int i=0; i<target.length; i++) {
            if (target[i] == val) {
                ret.add("Push");
            } else {
                while(target[i] != val && val <=n) {
                    ret.add("Push");
                    ret.add("Pop");
                    val++;
                }
                ret.add("Push");
            }
            val++;
        }

        return ret;
    }
}

File: 1552-build-an-array-with-stack-operations/1552-build-an-array-with-stack-operations.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        ret: List[str] = []
        count = 1
        for x in target:
            if x == count:
                ret.append("Push")
            else:
                while count != x:
                    ret.append("Push")
                    ret.append("Pop")
                    count += 1
                ret.append("Push")
            count+=1
        return ret
                